content/en/logs/explorer/calculated_fields/expression_language.md

```diff
@@ -0,0 +1,266 @@
+
+<h4>split_before(<i>str</i> string, <i>str</i> separator, <i>int</i> occurrence)</h4>
+
+Extracts the portion of text preceding a certain pattern in a string.
+
+<details>
+<summary>Example</summary>
+<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A log event has the following attribute: <code>@row_value</code> = "1,Bob,Smith"<br><br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Formula: split_before(<code>@row_value</code>, ",")
+<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Result: "1"
+</details>
+
+<h4>split_after(<i>str</i> string, <i>str</i> separator, <i>int</i> occurrence)</h4>
+
+Extracts the portion of text following a certain pattern in a string.
+
+<details>
+<summary>Example</summary>
+<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A log event has the following attributes: <code>@row_value</code> = "1,Bob,Smith"<br><br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Formula: split_after(<code>@row_value</code>, ",", 2)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Result: "Smith"
+</details>
+
+<h4>substring(<i>str</i> string, <i>int</i> start, <i>int</i> end)</h4>
+
+Extracts a portion of text from the middle of a string.
+
+<details>
+<summary>Example</summary>
+<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A log event has the following attributes: <code>@row_value</code> = "1,Bob,Smith"<br><br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Formula: substring(<code>@row_value</code>, 3, 3)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Result: "Bob"
+</details>
+
+<h4>suffix(<i>str</i> string, <i>int</i> num_chars)</h4>
+
+Extracts a portion of text from the end of a string.
+
+<details>
+<summary>Example</summary>
+<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A log event has the following attributes: <code>@url</code> = "www.datadoghq.com"<br><br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Formula: suffix(<code>@url</code>, 4)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Result: ".com"
+</details>
+
+<h4>textjoin(<i>str</i> delimiter, <i>expr</i> value [, <i>expr</i> value, …])</h4>
+
+Combines multiple values into a single string with a delimiter in between.
+
+<details>
+<summary>Example</summary>
+<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A log event has the following attributes:
+<br>&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;<code>@first_name</code> = "Bob"
+<br>&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;<code>@last_name</code> = "Smith"<br><br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Formula: textjoin(", ", <code>@last_name</code>, <code>@first_name</code>)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Result: "Smith, Bob"
+</details>
+
+<h4>upper(<i>str</i> string)</h4>
+
+Converts string to uppercase.
+
+<details>
+<summary>Example</summary>
+<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A log event has the following attributes: <code>@first_name</code> = "Bob"<br><br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Formula: upper(<code>@first_name</code>)
+<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Result: "BOB"
+</details>
+
+### Logical
+
+<h4>case(<i>expr</i> condition, <i>expr</i> value_if_true [, <i>expr</i> condition, <i>expr</i> value_if_true …], <i>expr</i> value_else)</h4>
+
+Evaluates a series of conditions and returns a value accordingly.
```

Comment:
Just highlighting that this is the only function that doesn't include an example. Not sure if that was intentional.